source/draw/gpu/opengl/backend/EglDisplayContext.ooc

```diff
@@ -56,6 +56,12 @@ EglDisplayContext: class {
 		this _generateContext(sharedContext ? sharedContext _eglContext : null, chosenConfig)
 	}
 	free: override func {
+		if (eglGetCurrentContext() != this _eglContext) {
+			if (eglGetCurrentContext() == EGL_NO_CONTEXT)
+				Debug error("Trying to unbind OpenGL context from a thread without a context!")
+			else
+				Debug error("Trying to unbind OpenGL context from another thread!")
+		}
```

Comment:
Updated with a check against freeing with a thread that has a context but not the same context.